### ISSUE TITLE
Should GenericWorks index their size?

### ISSUE THREAD

**Issue by grosscol** [projecthydra-labs/curation_concerns#528]
Ascertaining the size in bytes of the contents of a collection involves a solr query for each work in the collection to obtain the file_size of the contained FileSets.  Aggregating the file_sizes of the FileSets of a Work would remove one level of solr queries for calculating the size of a collection.

It was thought that this would make the size calculation require a single solr query.  However, since works can be nested, tree traversal is required anyway.

**Comment by awead** [comment]
Yes. We're doing it here: https://github.com/psu-stewardship/scholarsphere/issues/389

We can extract the size calculation method in CollectionBehavior and share it across both classes.
